src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/EvalResultDataItem.cs

```diff
@@ -253,7 +253,8 @@ private static DkmEvaluationResultFlags GetFlags(DkmClrValue value, DkmInspectio
                 }
             }
 
-            if (!value.IsError() && value.HasUnderlyingString(inspectionContext))
+            // check for truncated string before HasUndleryingString so we don't pull the string into memory just to set the flag
+            if (!value.IsError() && (value.EvalFlags.HasFlag(DkmEvaluationResultFlags.TruncatedString) || value.HasUnderlyingString(inspectionContext)))
```

Comment:
>value.EvalFlags.HasFlag(DkmEvaluationResultFlags.TruncatedString)  [](http://example.com/codeflow?start=37&length=66)

I think you want this check inside of `HasUnderlyingString` instead of here.